src/target/riscv/riscv-013.c

```diff
@@ -3530,12 +3530,13 @@ void riscv013_clear_abstract_error(struct target *target)
 
 #define COMPLIANCE_TEST(b, message) \
 {                                   \
+	LOG_INFO("Executing test %d (riscv-013.c:%d): %s", total_tests, __LINE__, message); \
```

Comment:
Why should we not replace `riscv-013.c` with `%s` format that uses `__FILE__` as a parameter?

Reply:
__FILE__ gives a (more or less) full path, which is rather long and yet does not add useful information. I therefore chose to abbreviate it to only the file name.

Reply:
Fair point. How about using the `strrchr()`, like so:

```suggestion
#ifdef _WIN32
#define FILE_SEP '\\'
#else
#define FILE_SEP '/'
#endif
#define COMPLIANCE_TEST(b, message) \
{\
	char *last_sep = strrchr(__FILE__, FILE_SEP);\
	char *fname = (last_sep == NULL ? __FILE__ : last_sep + 1);\
	LOG_INFO("Executing test %d (%s:%d): %s", total_tests, fname, __LINE__, message); \
```

Reply:
I don't feel that this (rather ugly) complexity is necessary for this use case; ultimately, I'd love to see the test have a meaningful name so that we don't need to rely on the line number.

Reply:
> (rather ugly) complexity

Beauty is in the eyes of the beholder :)

If the need to print the filename and line number is obviated, I'd agree with you. But if filename and line number are to be used, I'd rather rely on compiler to do the right thing automatically; that'd take care of the cases if this file is ever renamed, the macro is moved to a header and included by other `.c` files, etc.

Reply:
updated